tutorials/roofit/rf616_morphing.py

```diff
@@ -0,0 +1,138 @@
+
+        # Add the pdf to the workspace, the inOrder of 1 is necessary for calculation of the nll
+        # Adjust it to 0 to see binning
+        workspace.Import(ROOT.RooHistPdf(f"histpdf{i}", f"histpdf{i}", [x_var], hist, intOrder=0))
+
+        # Add the pdf to the grid and to the plot
+        grid.addPdf(workspace[f"histpdf{i}"], int(i))
+        workspace[f"histpdf{i}"].plotOn(frame1)
+
+    # Create the morphing and add it to the workspace
+    morph_func = ROOT.RooMomentMorphFuncND("morph_func", "morph_func", [mu_var], [x_var], grid, setting)
+
+    # Normalizes the morphed object to be a pdf, set it false to prevent warning messages and gain computational speed up
+    # morph_func.setPdfMode()
+
+    # Creating the morphed pdf
+    morph = ROOT.RooWrapperPdf("morph", "morph", morph_func, True)
+    workspace.Import(morph)
+    workspace["morph"].plotOn(frame1, LineColor="r")
+
+    c0 = ROOT.TCanvas()
+    frame1.Draw()
+
+    # The input prevents the plot from beeing closed instantly
+    input("")
+
+
+# Define the "observed" data in a workspade
+def build_ws(mu_observed):
+    x_var = ROOT.RooRealVar("x", "x", -5, 15)
+    mu_var = ROOT.RooRealVar("mu", "mu", mu_observed, 0, 4)
+    sigma_var = ROOT.RooRealVar("sigma", "sigma", 1.5)
+    gauss = ROOT.RooGaussian("gauss", "gauss", x_var, mu_var, sigma_var)
+    uniform = ROOT.RooUniform("uniform", "uniform", x_var)
+    obs_data = gauss.generate(x_var, n_samples)
+    obs_data.SetName("obs_data")
+
+    # Using a workspace for easier processing inside the class
+    workspace = ROOT.RooWorkspace()
+    workspace.Import(x_var)
+    workspace.Import(mu_var)
+    workspace.Import(gauss)
+    workspace.Import(uniform)
+    workspace.Import(obs_data)
+
+    return workspace
+
+
+# The "observed" data
+mu_observed = 2.5
+workspace = build_ws(mu_observed)
+x_var = workspace["x"]
+mu_var = workspace["mu"]
+sigma_var = workspace["sigma"]
+gauss = workspace["gauss"]
+uniform = workspace["uniform"]
+obs_data = workspace["obs_data"]
+
+
+# Create the exact negative log likelihood functions for Gaussian model
+nll_gauss = gauss.createNLL(obs_data)
+
+# Compute the morphed nll
+morphing(ROOT.RooMomentMorphFuncND.Linear)
+nll_morph = workspace["morph"].createNLL(obs_data)
+
+# Plot the negative logarithmic summed likelihood
+c1 = ROOT.TCanvas()
+frame = mu_var.frame(Title="SBI vs. Morphing")
+nll_gauss.plotOn(frame, LineColor="b", ShiftToZero=True, Name="gauss")
+nll_morph.plotOn(frame, LineColor="r", ShiftToZero=True, Name="morphed")
+frame.Draw()
+
+# Create a legend and add entries
+legend = ROOT.TLegend(0.7, 0.7, 0.9, 0.9) 
+legend.AddEntry("gauss", "Gaussian NLL", "l")
+legend.AddEntry("morphed", "Morphed NLL", "l")
+legend.Draw()
+
+
+# Compute the minimum via minuit and display the results
+for nll in [nll_gauss, nll_morph]:
+    min = minimizer = ROOT.RooMinimizer(nll)
+    minimizer.setErrorLevel(0.5)  # Adjust the error level in the minimization to work with likelihoods
```

Comment:
```suggestion
```
Not needed here, because the likelihood returned by createNLL has the right error definition.